controllers/login.js

```diff
@@ -103,46 +104,57 @@ var createSession = function(req, callback) {
   }, callback);
 };
 
-var setSessionCookie = function(res, cookie) {
-  var sessionId = SESSION_COOKIE_RE.exec(cookie)[1];
-  res.cookie('AuthSession', sessionId, { httpOnly: true });
+const getCookieOptions = req => {
+  return {
+    sameSite: 'lax', // prevents the browser from sending this cookie along with cross-site requests
+    secure: req.hostname !== 'localhost' // only transmit when requesting via https unless in development
```

Comment:
I wonder if this is spoofable. I.e., can you set the hostname header to whatever you like when sending a request?

If so, we could consider determining the hostname at application start (if this is reliable / known inside Amazon). Or perhaps introducing a development mode environment variable that lets us bypass this and perhaps other things later.